layer_metadata_restorer.go

```diff
@@ -13,29 +13,95 @@ import (
 
 //go:generate mockgen -package testmock -destination testmock/layer_metadata_restorer.go github.com/buildpacks/lifecycle LayerMetadataRestorer
 type LayerMetadataRestorer interface {
-	Restore(buildpacks []buildpack.GroupBuildpack, appMeta platform.LayersMetadata, cacheMeta platform.CacheMetadata) error
+	UseSHAFiles() bool
+	Restore(buildpacks []buildpack.GroupBuildpack, appMeta platform.LayersMetadata, cacheMeta platform.CacheMetadata, layerSHAStore LayerSHAStore) error
 }
 
 type DefaultLayerMetadataRestorer struct {
-	Logger     Logger
-	LayersDir  string
-	SkipLayers bool
+	logger      Logger
+	layersDir   string
+	skipLayers  bool
+	useShaFiles bool
 }
 
-func NewLayerMetadataRestorer(logger Logger, layersDir string, skipLayers bool) LayerMetadataRestorer {
+type LayerSHAStore interface {
+	add(buildpackID, sha string, layer *bpLayer) error
+	get(buildpackID string, layer bpLayer) (string, error)
+}
+
+type defaultLayerSHAStore struct {
+	useShaFiles              bool
```

Comment:
I wonder instead of having this boolean, we could have two different implementations of the interface - one using the filesystem, the other using memory. When the layer metadata restorer is constructed, we could pass in the right one.

Reply:
I like that suggestion @natalieparellano !

Reply:
I like it too! Please see https://github.com/buildpacks/lifecycle/pull/624/commits/76b95b3821257dd6127e05958f303dcab437fab8.